nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-headless-server/src/main/java/org/apache/nifi/headless/HeadlessNiFiServer.java

```diff
@@ -165,6 +168,16 @@ public void preDestruction() throws AuthorizerDestructionException {
             FlowManager flowManager = flowController.getFlowManager();
             flowManager.getGroup(flowManager.getRootGroupId()).startProcessing();
 
+            final NarLoader narLoader = new StandardNarLoader(
```

Comment:
So I tried it but was not able to make it work. I checked and in NiFi we have this property defined which is missing from MiNiFI if I'm not mistaken:
nifi.nar.library.autoload.directory=./extensions

After some investigation it turned out that it doesn't help if I update the properties file as it is generated based on the ConfigTransformer. I tried to add the config there but even with that if a place  anar into the extensions dir I don't see any logs stating that the nar is picked up. Maybe something is missing on my side but is it possible that some configuration is missing from the PR?

Reply:
I think this was because you already had the NAR in `./extensions` so it got picked up by the normal loading stuff.